py/torch_tensorrt/dynamo/conversion/impl/quantize.py

```diff
@@ -0,0 +1,46 @@
+from typing import Optional
+
+import numpy as np
+import tensorrt as trt
+from torch.fx.node import Target
+from torch_tensorrt.dynamo._SourceIR import SourceIR
+from torch_tensorrt.dynamo.conversion._ConversionContext import ConversionContext
+from torch_tensorrt.dynamo.conversion.converter_utils import get_trt_tensor
+from torch_tensorrt.fx.converters.converter_utils import set_layer_name
+from torch_tensorrt.fx.types import TRTTensor
+
+
+def quantize_fp8(
+    ctx: ConversionContext,
+    target: Target,
+    source_ir: Optional[SourceIR],
+    name: str,
+    input_tensor: TRTTensor,
+    scale: np.ndarray,
+) -> TRTTensor:
+    """
+    Adds quantize and dequantize ops (QDQ) which quantize to INT8 or FP8 based
+    on the output_type set and dequantizes them back.
+    """
+    if (isinstance(input_tensor, TRTTensor)) and not (
+        input_tensor.dtype != trt.float32 or input_tensor.dtype != trt.float16
+    ):
+        raise ValueError(
+            f"quantize_fp8 converter received an input of {input_tensor.dtype} type. Supported types: float32 | float16"
+        )
+
+    if isinstance(scale, np.ndarray):
+        scale = get_trt_tensor(ctx, scale, name + "_scale")
```

Comment:
Can `scale` also be a TRT tensor? If so, can the function header schema be changed to `Union[np.ndarray, TRTTensor]`

Reply:
scale can only be constants.

Reply:
Can the `if isinstance` be removed then in that case, or are there situations where the `scale` might not be a numpy array?